dhis-2/dhis-support/dhis-support-commons/src/main/java/org/hisp/dhis/commons/collection/CollectionUtils.java

```diff
@@ -291,4 +293,16 @@ public static <K, V> Map<K, V> mapOf(
             Map.entry( k10, v10 ),
             Map.entry( k11, v11 ) );
     }
+
+    /**
+     * Convert an Iterator to a Stream.
+     *
+     * @param iterator The Iterator to convert to a stream.
+     * @return A stream of the iterable.
+     */
+    public static <E> Stream<E> iterableToStream( Iterator<E> iterator )
+    {
+        Iterable<E> iterable = () -> iterator;
+        return StreamSupport.stream( iterable.spliterator(), false );
```

Comment:
Better use : 
```java
StreamSupport.stream(spliteratorUnknownSize(iter, Spliterator.ORDERED | Spliterator.NONNULL | Spliterator.IMMUTABLE), false);
```
`Spliterator.NONNULL` might not apply.

Reply:
Ok